encoders.go

```diff
@@ -0,0 +1,97 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Contributor: Julien Vehent jvehent@mozilla.com [:ulfr]
+
+package main
+
+import (
+	"encoding/asn1"
+	"encoding/base64"
+	"fmt"
+	"math/big"
+	"strings"
+)
+
+func encode(sig *ecdsaSignature, format string) (str string, err error) {
+	if len(format) < 4 {
+		format = "rs_base64url"
+	}
+	if format[0:9] == "rs_base64" {
+		// return default rs_base64url format
+		rs := make([]byte, len(sig.R.Bytes())+len(sig.S.Bytes()))
+		copy(rs[:len(sig.R.Bytes())], sig.R.Bytes())
+		copy(rs[len(sig.S.Bytes()):], sig.S.Bytes())
+		str = base64.StdEncoding.EncodeToString(rs)
+		if format == "rs_base64url" {
+			str = b64Tob64url(str)
+		}
+		return str, nil
+	}
+	if format[0:10] == "der_base64" {
```

Comment:
Good point. I never think about using, because doing it by hand it so straightforward. Fwiw, the `HasPrefix` implementation does exactly the same thing:
```go
   373	func HasPrefix(s, prefix string) bool {
   374		return len(s) >= len(prefix) && s[0:len(prefix)] == prefix
   375	}
```